From 52c739bc081adcab4e34c6840923d6722e394829 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Thu, 11 Jun 2020 19:46:41 +0200
Subject: [PATCH] strip fno-plt which rootcling does not understand

---
 src/etc/dictpch/makepch.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/etc/dictpch/makepch.py b/src/etc/dictpch/makepch.py
index ddba7d9..e72221d 100755
--- a/src/etc/dictpch/makepch.py
+++ b/src/etc/dictpch/makepch.py
@@ -90,6 +90,9 @@ def makepch():
    if "-isystem" in cppFlags:
       idx = cppFlags.index("-isystem")
       del cppFlags[idx: idx+2]
+   if "-fno-plt" in cppFlags:
+      idx = cppFlags.index("-fno-plt")
+      del cppFlags[idx: idx+1]
 
    cppflagsList=["-D__CLING__",
                  "-D__STDC_LIMIT_MACROS",
-- 
2.27.0

