From 37e4d9015dd399143476d1444ef8b5426f377088 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sun, 20 Jun 2021 13:38:15 +0200
Subject: [PATCH] LLVM_DEFS are not available when we do not build llvm

---
 src/interpreter/CMakeLists.txt | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/src/interpreter/CMakeLists.txt b/src/interpreter/CMakeLists.txt
index 217a3d5..c7f98b8 100644
--- a/src/interpreter/CMakeLists.txt
+++ b/src/interpreter/CMakeLists.txt
@@ -445,16 +445,19 @@ endif()
 # add_custom_target(CLING)
 set(CLING_LIBRARIES clingInterpreter;clingMetaProcessor;clingUtils CACHE STRING "")
 if (builtin_cling)
-
-  # Extract the compilation flags from LLVM and make them public to the
-  # rest of ROOT so that we can compile against LLVM with matching flags.
-
-  # LLVM doesn't really give us a API to get this with an in-source build
-  # so we just use the normal way of doing this and read the llvm directory
-  # compilation properties.
-  get_directory_property(LLVM_DEFS DIRECTORY llvm/src COMPILE_DEFINITIONS)
-  # Turns DEFINE1;DEFINE2 to -DDEFINE1 -DDEFINE2
-  string (REPLACE ";" " -D" LLVM_DEFS ";${LLVM_DEFS}")
+  if (builtin_llvm)
+    # Extract the compilation flags from LLVM and make them public to the
+    # rest of ROOT so that we can compile against LLVM with matching flags.
+
+    # LLVM doesn't really give us a API to get this with an in-source build
+    # so we just use the normal way of doing this and read the llvm directory
+    # compilation properties.
+    get_directory_property(LLVM_DEFS DIRECTORY llvm/src COMPILE_DEFINITIONS)
+    # Turns DEFINE1;DEFINE2 to -DDEFINE1 -DDEFINE2
+    string (REPLACE ";" " -D" LLVM_DEFS ";${LLVM_DEFS}")
+  else()
+    set(LLVM_DEFS "")
+  endif()
 
   # FIXME: Reduce the usage of CLING_CXXFLAGS by adding a cmake routine in
   # RootMacros.cmake for all cling-dependent libraries
-- 
2.30.2

